ganttproject/src/biz/ganttproject/impex/csv/XlsWriterImpl.java

```diff
@@ -20,45 +20,65 @@
 package biz.ganttproject.impex.csv;
 
 import com.google.common.base.Preconditions;
+
+import org.apache.poi.hssf.usermodel.HSSFDataFormat;
 import org.apache.poi.hssf.usermodel.HSSFWorkbook;
 import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellStyle;
 import org.apache.poi.ss.usermodel.Row;
 import org.apache.poi.ss.usermodel.Sheet;
 import org.apache.poi.ss.usermodel.Workbook;
 
 import java.io.IOException;
 import java.io.OutputStream;
+import java.math.BigDecimal;
+import java.util.Calendar;
 
 /**
  * @author akurutin on 04.04.2017.
  */
 public class XlsWriterImpl implements SpreadsheetWriter {
   private final Workbook myWorkbook;
-  private final Sheet mySheet;
   private final OutputStream myStream;
+  private final CellStyle myDateStyle;
+  private final CellStyle myIntegerStyle;
+  private final CellStyle myDoubleStyle;
 
-
+  private Sheet myCurrentSheet = null;
   private Row myCurrentRow = null;
   private int myNextRowInd = 0;
   private int myNextCellInd = 0;
 
 
-  XlsWriterImpl(OutputStream stream) {
+  XlsWriterImpl(OutputStream stream, String initialSheetName) {
     myStream = Preconditions.checkNotNull(stream);
     myWorkbook = new HSSFWorkbook();
-    mySheet = myWorkbook.createSheet();
+
+    myDateStyle = myWorkbook.createCellStyle();
+    short fmt = myWorkbook.createDataFormat().getFormat("yyyy-mm-dd");
+    myDateStyle.setDataFormat(fmt);
+
+    // https://poi.apache.org/apidocs/org/apache/poi/ss/usermodel/BuiltinFormats.html
+    myIntegerStyle = myWorkbook.createCellStyle();
+    myIntegerStyle.setDataFormat((short) 1);
+    myDoubleStyle = myWorkbook.createCellStyle();
+    myDoubleStyle.setDataFormat((short) 2);
+
+    myCurrentSheet = myWorkbook.createSheet(initialSheetName);
   }
 
   @Override
   public void print(String value) throws IOException {
-    if (myCurrentRow == null) {
-      createNewRow();
+    if (value != null) {
```

Comment:
is it possible that value is nullable? What does it mean from exporter's point of view? If it is empty cell then we should probably add a cell anyway, just don't fill it with value.